qiskit/algorithms/gradients/spsa_estimator_gradient.py

```diff
@@ -0,0 +1,103 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Gradient of Sampler with Finite difference method."""
+
+from __future__ import annotations
+
+from typing import Sequence
+
+import numpy as np
+
+from qiskit.circuit import Parameter, QuantumCircuit
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .base_estimator_gradient import BaseEstimatorGradient
+from .estimator_gradient_result import EstimatorGradientResult
+
+
+class SPSAEstimatorGradient(BaseEstimatorGradient):
+    """
+    Compute the gradients of the expectation value by the Simultaneous Perturbation Stochastic
+    Approximation (SPSA).
+    """
+
+    def __init__(
```

Comment:
We should add a `batch_size` here which allows to average multiple SPSA gradients into one. That would be used in ML and SPSA optimizations 🙂

Reply:
I added `batch_size` option.